src/osd/PG.cc

```diff
@@ -5531,6 +5532,7 @@ void PG::handle_advance_map(
   recovery_state.handle_event(evt, rctx);
   if (pool.info.last_change == osdmap_ref->get_epoch())
     on_pool_change();
+  update_store_with_options();
```

Comment:
the only other place we'd need to do this is from load_pgs(), or whatever the loaded() method is that it calls